policy/modules/kernel/kernel.if

```diff
@@ -2894,6 +2948,40 @@ interface(`kernel_dontaudit_read_unlabeled_files',`
 	dontaudit $1 unlabeled_t:file { getattr read };
 ')
 
+########################################
+## <summary>
+##	Create an object in unlabeled directories
+##	with a private type.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+## <param name="private type">
+##	<summary>
+##	The type of the object to be created.
+##	</summary>
+## </param>
+## <param name="object">
+##	<summary>
+##	The object class of the object being created.
+##	</summary>
+## </param>
+## <param name="name" optional="true">
+##	<summary>
+##	The name of the object being created.
+##	</summary>
+## </param>
+#
+interface(`kernel_unlabeled_filetrans',`
```

Comment:
This is horrifying too. Why doesn't the fs have labels?

Reply:
I wonder the same. It seems to be a bug in Docker, though I can't say that for certainty.

The behavior I observed is that if Docker is built with `btrfs` support, it will use btrfs subvolumes for storing container images. These subvolumes get created without a `rootcontext=` option, making the subvolume root unlabeled. Without this file type transition, all the files inside this volume will also be unlabeled when the container image is extracted there.

For some reason, when Docker creates new containers from this image, it will also use a btrfs subvolume, except that volume will be labeled correctly, MCS categories and all. This behavior doesn't occur if Docker is not using btrfs subvolumes.

This doesn't seem to be a problem in `fedora-selinux` because their container engine/runtime type is both unconfined and has access to unlabeled files.

With the above access to `docker_t`, the end result is that the files inside these subvolumes will be labeled correctly but the subvolume root mountpoint will still be unlabeled. I haven't found a proper solution for that.

Reply:
Ok, then I guess we're stuck with it for now. My request would be for you to file a bug on docker, if not do a patch to fix it. Not tied to merging this PR, but it would be nice to move Docker forward.

Reply:
I've opened https://github.com/moby/moby/issues/43088 regarding this.